### PR TITLE
Fix native IME input at the beginning of a line

### PR DESCRIPTION
**Pull request by blakeembrey**
Closes https://github.com/ianstormtaylor/slate/issues/1340.
Related to https://github.com/ianstormtaylor/slate/pull/1322.

/cc @doodlewind, did you want to test your non-native pinyin input first? This works for the native macOS input.

### CODE REVIEW

File: packages/slate-react/src/components/leaf.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ class Leaf extends React.Component {
     // COMPAT: If the text is empty and it's the only child, we need to render a
     // line break to get the block to have the proper height.
     if (text == '' && parent.kind == 'block' && parent.text == '') {
-      return <span data-slate-empty-block>{'\n'}</span>
+      return <span data-slate-empty-block>{'\u200B'}</span>
```
Comment (doodlewind) [inline packages/slate-react/src/components/leaf.js]: This works! For the remaining two failing test case, just use

```
<span data-slate-empty-block="true"></span>
```

instead of

```
<span data-slate-empty-block="true">\n</span>
```

and I believe that'll be adequate.

**Comment by doodlewind** [comment]
Third party pinyin IME works fine still, good job @blakeembrey !

File: packages/slate-react/src/components/leaf.js (hunk 1 above)
Comment (blakeembrey) [reply packages/slate-react/src/components/leaf.js]: Good catch! Forgot to run the tests 😄

**Comment by blakeembrey** [comment]
Just a note, but I'm still working on this. I think I ran into some edge cases that can happen and found a cleaner way around all this.

**Comment by blakeembrey** [comment]
Ok, managed to kill a bunch of code and fix the functionality all around. Appreciate if you test it again @doodlewind using your custom IME input since I only have the macOS default included.

I killed the `data-slate-empty-block` span because it made things more complicated - it actually had been missed about when copying and pasting. There's special code that removed `data-slate-zero-width` but not the new empty element. Combining them back together makes it easier to avoid that mistake.

I also killed the code moving the offset in `findDOMPoint` since it gets corrected in `findPoint` anyway. All that was required to handle IME input was instead subtracting 1 from the size of the zero width element instead of resetting it back to 0. This allows the browser to type in the empty span while using IME and then fix it up later.

**Comment by ianstormtaylor** [comment]
This is awesome @blakeembrey! Love the code removal. I'll wait for the testing on other IMEs before merging. Looks good for regular stuff from my own testing though.

**Comment by ianstormtaylor** [comment]
@doodlewind no idea! 😄 Looks good to me though.

**Comment by doodlewind** [comment]
@ianstormtaylor Done!

**Comment by blakeembrey** [comment]
@doodlewind What does "Replace Multi Blocks" mean and does the cross mean it's failing today (and was it before)?

**Comment by doodlewind** [comment]
@blakeembrey This is not a regression, but a known issue in #967 I think.

**Comment by ianstormtaylor** [comment]
@blakeembrey @doodlewind so this is good to merge?

**Comment by blakeembrey** [comment]
@ianstormtaylor Looks like it! Exciting 😄

**Comment by doodlewind** [comment]
@ianstormtaylor I've tested and feeling it's good. 😁 To my knowledge no regression introduced.

**Comment by ianstormtaylor** [comment]
Thanks you two!
